uptown_funk/src/lib.rs

```diff
@@ -121,13 +127,46 @@ impl<S, E: Executor> StateWrapper<S, E> {
         self.state.borrow_mut()
     }
 
+    pub fn get_state(&self) -> Rc<RefCell<S>> {
+        self.state.clone()
+    }
+
     pub fn executor(&self) -> &E {
         &self.env
     }
 
     pub fn memory(&self) -> memory::Memory {
         self.env.memory()
     }
+
+    pub fn recover_state(self) -> Result<S, ()> {
+        match Rc::try_unwrap(self.state) {
+            Ok(s) => Ok(s.into_inner()),
+            Err(_) => Err(()),
+        }
+    }
+}
+
+unsafe impl<S, E: Executor> Send for StateWrapper<S, E> {}
+unsafe impl<S, E: Executor> Sync for StateWrapper<S, E> {}
```

Comment:
This is only required by Wasmer and can be marked with: `#[cfg(feature = "vm-wasmer")]`.
`StateWrapper` is obviously `!Send` and `!Sync` because of `Rc<RefCell>`;

For now this is ok, but could be an issue later. Not sure what the best approach here is.